datasets/snips_built_in_intents/snips_built_in_intents.py

```diff
@@ -0,0 +1,142 @@
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "text": datasets.Value("string"),
+                    "label": datasets.features.ClassLabel(
+                        names=[
+                            "ComparePlaces",
+                            "RequestRide",
+                            "GetWeather",
+                            "SearchPlace",
+                            "GetPlaceDetails",
+                            "ShareCurrentLocation",
+                            "GetTrafficInformation",
+                            "BookRestaurant",
+                            "GetDirections",
+                            "ShareETA",
+                        ]
+                    ),
+                }
+            ),
+            homepage="https://github.com/sonos/nlu-benchmark/tree/master/2016-12-built-in-intents",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        # Note: The source dataset doesn't have a train-test split.
+        # ToDo: Consider splitting the data into train-test sets and re-hosting.
+        samples_path = dl_manager.download_and_extract(_DOWNLOAD_URL)
+        return [
+            datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"filepath": samples_path}),
+        ]
+
+    def _generate_examples(self, filepath):
+        """Snips built in intent examples."""
+        samples_txt_lbl = []  # type: List[Tuple[str, str]]
+        num_examples = 0
+
+        with open(filepath, encoding="utf-8") as file_obj:
+            snips_dict = json.load(file_obj)
+            domains = snips_dict["domains"]
+
+            for domain_dict in domains:
+                intents = domain_dict["intents"]
+
+                for intent_dict in intents:
+                    # ToDo: The file format is not expected to change. Is it ok to throw on key errors here?
+                    label = intent_dict["benchmark"]["Snips"]["original_intent_name"]
+                    queries = intent_dict["queries"]
+
+                    for query_dict in queries:
+                        query_text = query_dict["text"]
+
+                        samples_txt_lbl.append(
+                            (query_text, label)
+                        )  # Make a list of all 328 examples, then shuffle and yield below.
+
+                        # Option 1 (Unshuffled): Yield samples here if un-shuffled samples ok.
+                        # yield num_examples, {"text": query_text, "label": label}
+
+                        num_examples += 1  # Explicitly keep track of the number of examples.
+
+        # Option 2 (Shuffled): Yield samples here if shuffled samples are required.
+        random.Random(12345).shuffle(
+            samples_txt_lbl
+        )  # Use a new instance of Random to fix shuffle pattern without affecting global rand.
```

Comment:
could yo not shuffle the data ?

If the user wants to shuffle they can do `dataset.shuffle()`

Reply:
Hi @lhoestq . I was hoping to have this discussion :-)
Yes, I don't need to shuffle. There is an 'option 1' a little higher up in the code that would then be the way to go. I'll uncomment that option and ping you again.

Reply:
@lhoestq You can have a look. If you are happy with the current version then we can remove all of the commented code.